### DISCUSSION TITLE
print history and depth via ros2 topic info -v

### DISCUSSION THREAD

**Discussion by jelly-FF** [eProsima/Fast-DDS#3682] [General]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

`ros2 topic info -v` show all the qos profile correctly

### Current behavior

Most of the qos profile are shown correctly, except History (Depth). 
History (Depth) should not be "UNKNOWN".

ROS2 replied the reason is that Fast-DDS does not set QoS history nor depth during endpoint discovery. [ros2/rclpy#849(comment)](https://github.com/ros2/rclpy/pull/849#issuecomment-988548722)

![image](https://github.com/ros2/ros2/assets/33385430/b6d430bd-e513-4d36-8095-d006abcbd532)

### Steps to reproduce

1. run ros2 minimal_publisher
```
ros2 run examples_rclcpp_minimal_publisher publisher_member_function
```

2. check qos profile
```
ros2 topic info -v /topic
```

### Fast DDS version/commit

- ROS2 Version or commit hash:
  - Humble
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - ros2cli

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

Ubuntu 22.04

### XML configuration file

not used

**Comment by JesusPoderoso** [comment]
Hi @jelly-FF, thanks for your report
According to [our contributing guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md), I am moving this PR to the proper discussion forum.
Following the standard, in section 2.2.4 it is explained that the History QoS is not RxO (Requested / Offered) mandatory.
That means that History QoS is not mandatory to be included in the data transmission.
Fast DDS includes all RxO mandatory QoS, and some other not RxO mandatatory QoS, but not the History QoS.

**Reply by jelly-FF** [reply]
Is there any quick fix I can try in source code to include History QoS?

**Reply by JesusPoderoso** [reply]
Unfortunately there is no quick fix.
That would involve including the history QoS in several layers and communication steps, and also in the RMW for ROS purposes.
